tests/infrastructure/test_ip_cache.py

```diff
@@ -0,0 +1,225 @@
+"""Test that the IP cache is working correctly. (No false positives, no collisions, speed, etc.)."""
+from __future__ import annotations
+
+import pytest
+
+import numpy as np
+import os
+import time
+from copy import deepcopy
+from pathlib import Path
+from qonnx.core.datatype import DataType
+from qonnx.custom_op.registry import getCustomOp
+from qonnx.transformation.general import GiveReadableTensorNames, GiveUniqueNodeNames
+from qonnx.util.basic import gen_finn_dt_tensor
+from typing import TYPE_CHECKING, Literal, cast
+
+from finn.custom_op.fpgadataflow.hls.matrixvectoractivation_hls import MVAU_hls
+from finn.custom_op.fpgadataflow.hlsbackend import HLSBackend
+from finn.custom_op.fpgadataflow.rtl.matrixvectoractivation_rtl import MVAU_rtl
+from finn.custom_op.fpgadataflow.rtlbackend import RTLBackend
+from finn.transformation.fpgadataflow.ip_cache import CachedIPGen, IPCache
+from finn.transformation.fpgadataflow.specialize_layers import SpecializeLayers
+from finn.util.basic import alveo_part_map
+from finn.util.deps import get_cache_path
+from tests.fpgadataflow.test_fpgadataflow_mvau import make_single_fclayer_modelwrapper
```

Comment:
This import doesn't work, at least in the CI where FINN+ is launched as a pip-installed package.